server/store.go

```diff
@@ -547,8 +547,7 @@ func (sm *StoreMsg) clear() {
 	if sm == nil {
 		return
 	}
-	*sm = StoreMsg{_EMPTY_, nil, nil, sm.buf, 0, 0}
-	if len(sm.buf) > 0 {
-		sm.buf = sm.buf[:0]
-	}
+	sm.subj = _EMPTY_
+	sm.hdr, sm.msg, sm.buf = sm.hdr[:0], sm.msg[:0], sm.buf[:0]
```

Comment:
The comment at top of function was saying that all fields are nil/empty, except for buf if present but reset to [:0]. I honestly don't know the implications of sm.hdr[:0] as opposed to set it to nil (same for sm.msg), in that if caller checks (incorrectly?) nil vs len(hdr)==0 to decide what to do, then this is a change. Again, correct code should not care and so that should not make a difference, but again, not sure. If that does not change your profiling, maybe keep those at nil?

Reply:
I can’t be sure there aren’t checks for `nil` precisely, but that would be quite surprising if so. I’ve been running the branch today and it seems to be OK admittedly.

Reply:
Will see if @derekcollison thinks that this could be an issue or not.